modules/islandora_wos/islandora_wos.module

```diff
@@ -0,0 +1,139 @@
+      'title' => 'Web of Science',
+      'description' => 'Configure Web of Science integration.',
+      'page callback' => 'drupal_get_form',
+      'page arguments' => array('islandora_wos_admin_form'),
+      'access arguments' => array('administer site configuration'),
+      'type' => MENU_LOCAL_TASK,
+      'file' => 'includes/admin.form.inc',
+    ),
+  );
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function islandora_wos_block_info() {
+  return array(
+    'islandora_wos_badge' => array(
+      'visibility' => BLOCK_VISIBILITY_LISTED,
+      'pages' => 'islandora/object/*',
+      'cache' => DRUPAL_CACHE_PER_PAGE,
+      'info' => t('Islandora Web of Science Badge'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function islandora_wos_block_view($delta = '') {
+  module_load_include('inc', 'islandora_badges', 'includes/utilities');
+  $to_render = array();
+
+  if ($delta == 'islandora_wos_badge') {
+    // Get the WoS AMR username and password from the admin config.
+    $wos_username = variable_get('islandora_wos_username', '');
+    $wos_password = variable_get('islandora_wos_password', '');
+    if (!empty($wos_username) && !empty($wos_password)) {
+      $object = menu_get_object('islandora_object', 2);
+      if ($object) {
+        $doi = islandora_badges_get_doi($object);
+        if ($doi) {
+          // Embed Web of Science.
+          // Set API endpoint URL.
+          // @TODO: Add to admin form.
+          $url = "https://ws.isiknowledge.com/cps/xrpc";
+
+          // Store XML Request in string, as I can't send an XML file by POST.
+          $input_xml = '<?xml version="1.0" encoding="UTF-8" ?>
+      <request xmlns="http://www.isinet.com/xrpc42" src="app.id=API Demo">
+        <fn name="LinksAMR.retrieve">
+          <list>
+            <!-- WHO IS REQUESTING -->
+            <map>   
+              <val name="username">' . $wos_username . '</val>       
+              <val name="password">' . $wos_password . '</val>            
+            </map>
+            <!-- WHAT IS REQUESTED -->
+            <map>
+              <list name="WOS">
+                <val>timesCited</val>
+                <val>citingArticlesURL</val>
+              </list>
+            </map>
+            <!-- LOOKUP DATA -->
+            <map>  
+              <map name="cite_1">
+                <val name="doi">' . $doi . '</val>
+              </map>
+            </map>
+          </list>
+        </fn>
+      </request>';
+
+          // Post the request and get results!
+
+          $result = drupal_http_request($url, array(
+            'headers' => array('Content-Type' => 'text/xml'),
+            'data' => $input_xml,
+            'method' => 'POST',
+            'timeout' => 10,
+            )
+          );
+
+          // Convert the response to an array and remove the CDATA tags.
+
+          $wos_result = simplexml_load_string($result->data, 'SimpleXMLElement', LIBXML_NOCDATA);
```

Comment:
I think the community's unwritten rule as of late has been to avoid `SimpleXML` like the plague as it's given us nothing but trouble with encoding issues. Using a combination of `DOMDocument` and `DOMXpath` would be more verbose, but it would also let you dump that `switch` statement and some of the iteration/checking of `[0]`, instead checking `DOMXPath::query()->length` to see whether anything came back.